### PR TITLE
(1312) Refactor away the status field for activities

### PR DESCRIPTION
**Pull request by CristinaRO**
## Changes in this PR

- IATI status is calculated on the fly from the programme status

## Screenshots of UI changes

N/A

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

File: app/models/activity.rb
Comment (leeky) [inline app/models/activity.rb:368]: As this is only used in the IATI XML view, is `ActivityPresenter` a better place for it - maybe renamed to `iati_status`?

File: spec/models/activity_spec.rb
Comment (leeky) [inline spec/models/activity_spec.rb:1406]: Aren't these already covered by `ProgrammeToIatiStatus` tests?

Comment (CristinaRO) [reply spec/models/activity_spec.rb:1406]: Yep, but this was when I wasn't sure we'd be keeping `ProgrammeToIatiStatus`, pending my questions to Meyric. As it is, I think I'll wait until #843 is merged, then use the new `load_yaml` method to do that, possibly in the presenter like you suggested, and then remove the service object after all.

File: app/models/activity.rb
Comment (CristinaRO) [reply app/models/activity.rb]: I've renamed it, but it turns out that we don't use the presenter to show the activity in XML format, so it's still in the model.

**Review by leeky — APPROVED** [review]
LGTM! :shipit:

File: spec/models/activity_spec.rb
Comment (leeky) [reply spec/models/activity_spec.rb]: Good call! The YAML files are the right place to store these BEIS-approved mappings 🎉

**Pull request merged**
